.claude/owasp-mobile-security-checker/scripts/analyze_storage_security.py

```
@@ -0,0 +1,250 @@
+def scan_shared_preferences_usage(file_path: Path) -> List[Dict]:
+    """Scan for insecure SharedPreferences usage."""
+    findings = []
+
+    try:
+        with open(file_path, 'r', encoding='utf-8', errors='ignore') as f:
+            content = f.read()
+            lines = content.split('\n')
+
+            # Check for SharedPreferences import
+            has_shared_prefs = 'package:shared_preferences' in content
+            has_secure_storage = 'package:flutter_secure_storage' in content
```

Ruff (0.15.1)

[error] Local variable `has_secure_storage` is assigned to but never used

Remove assignment to unused variable `has_secure_storage`

(F841)